### ISSUE TITLE
UDS packaging standards / norms

### ISSUE THREAD

**Issue by Racer159** [defenseunicorns/uds-package-dubbd#395]
Blame @oates for this issue:

Perusing the repo looking for patterns that could be shared or cleaned up I noticed that there are some antipatterns or non-patterns that seem to exist within it:

For Example:
- https://github.com/defenseunicorns/uds-package-dubbd/blob/main/aws/uds-core-aws/zarf.yaml#L255C1-L255C124 - should have a `./` in front of it (all calls to zarf from within actions should) so that you don't need Zarf on your path and so that you don't run into version mismatches if you have multiple Zarf versions on the system
- https://github.com/search?q=repo%3Adefenseunicorns%2Fuds-package-dubbd%20awk&type=code - all of the `awk ###VAR### | sed` files should be removed in favor of `autoIndent` since that is much cleaner

I am sure there are more but that was what I noticed initially - happy to have a conversation / code review to establish common patterns that can be reused and potentially shared with UDS and others.

**Comment by Racer159** [comment]
For context on Zarf we enforce code QL linting and generally follow the Uber style guide for Go: https://github.com/uber-go/guide/blob/master/style.md

**Comment by mjnagel** [comment]
For the second part regarding `autoIndent` and `sed`/`awk` I'm not sure I totally follow? The cat_cert/key scripts are turning a zarf var with a filename into a zarf var containing the contents of that file. Is there a pattern for doing that with `autoIndent` (maybe its just a simpler action of `cat <filename>` and `autoIndent` for the `setVariables`?

**Comment by Racer159** [comment]
Correct - it would be simpler and be more understandable to others coming into the project and once https://github.com/defenseunicorns/zarf/pull/1906 is merged / released may not be needed anymore.

**Comment by mjnagel** [comment]
Don't want to increase the scope of this issue/PR too much so focusing on these things with the linked PR:
- Initial issues with `zarf` v `./zarf` and the handling of files to vars with sed/awk (second one can be reworked when we want to adopt the new `type: files` from zarf 0.28.3)
- CI workflows to perform linting using `yamllint` since *most* of our "code" is yaml
- Large batch of changes to fix linting issues
- General cleanup to remove invalid comments, extra files, etc to de-clutter

I think there's _a lot_ of ways to improve here still though:
- additionally linting/checking on terraform in this repo - especially if we are bringing more tf directly into this repo
- shellcheck/similar on the actions in our zarf packages - maybe lower priority
- validation of zarf.yamls against schema - https://github.com/defenseunicorns/uds-package-dubbd/issues/284
- evaluate pre-commit things vs CI things
